### PR TITLE
TM-4966, Publishable Position Ui

### PR DESCRIPTION
**Pull request by M00staff**
Working with all the integration that's been added so far - the edit dropdowns use hardcoded data... also, the position classification stuff is gonna be blank (can hardcode that for now too - but i dont think its necessary as that's already been demoed

https://metaphase.atlassian.net/browse/TM-4966

image

### CODE REVIEW

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 1]
```diff
@@ -24,32 +38,61 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
...
+      { 'Tour of Duty': data?.status || DEFAULT_TEXT },
+      { 'Pay Plan': '---' },
+      { 'Functional Bureau': 'None Listed' },
+      { 'Assignee': data?.status || DEFAULT_TEXT },
+      { 'Post Differential | Danger Pay': getDifferentials(pos) },
     ],
     textarea: data?.positionDetails || 'No description.',
     metadata: [
-      { 'Last Updated': formatDateFromStr(data?.lastUpdated) },
+      { 'Last Updated': (updateDate && updateUser) ? `${updateUser} ${updateDate}` : (updateDate || NO_UPDATE_DATE) },
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:56]: I would keep this as it was before as it is further along in the integration stage - like currently only the user ID is being passed from WS, so we wouldnt show that - and the date is a new format, long string, which we have to format to show

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 2]
```diff
@@ -97,10 +180,49 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
               />
             </Linkify>
             <div className="word-count">
-              {textArea.length} / 2000
+              {textArea.length} / 4,000
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:183]: intentionally capped at 2k after integration

**Comment by elizabeth-jimenez** [comment]
We'll want to revert all of the redux and action file changes, as this feature is now successfully integrated - just partially so is the deal.

What we want to do for this PR is really just add back the FE stuff we removed and really you can just hard code it on a single card just to call the UI G2G

**Comment by elizabeth-jimenez** [comment]
Basically: If we removed it for 11.1, then we likely want to add it back in, but if we changed it and kept it in, then that stuff is already production ready (and in prd).

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 3]
```diff
@@ -45,6 +56,22 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
...
+    dropdownHeight: 200,
+    includeSelectAll: true,
+    renderList: renderSelectionList,
+    className: 'width-280',
+  };
+  const [status, setStatus] = useState({});
+  const [exclude, setExclude] = useState(true);
+  const [selectedCycles, setSelectedCycles] = useState([]);
+  const [selectedFuncBureau, setSelectedFuncBureau] = useState('');
+  const [overrideTOD, setOverrideTOD] = useState('');
```
Comment (M00staff) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:72]: this is all stuff for the dropdowns in the edit mode

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 4]
```diff
@@ -124,6 +229,59 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
...
+    setFormData(newFormData);
+  };
+
+  const classificationTable = () => (
+    <div className="position-classifications">
+      <div className="line-separated-fields">
+        <div>
+          <span>Position:</span>
+          {/* <span>{bureau} {positionNumber}</span> */}
+          <span>{'AF'} {'12345'}</span>
```
Comment (M00staff) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:251]: just commenting out for now until the integration for Position Classification is done

**Comment by M00staff** [comment]
faeature flags added!

File: src/Components/Assignments/NotificationCard/Tabs/Assignments.jsx [hunk 5]
```diff
@@ -38,7 +38,6 @@ const Assignments = () => {
     const { destination } = result;
     // dropped outside the list
     if (!destination) {
-      console.log('hello world');
```
Comment (elizabeth-jimenez) [inline src/Components/Assignments/NotificationCard/Tabs/Assignments.jsx:41]: removed cause i shoudla removed it on the other PR

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 6]
```diff
@@ -27,14 +41,16 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
...
     ],
-    bodySecondary: [
-      // { 'Bid Cycle': data?.status || DEFAULT_TEXT },
-      // { 'TED': data?.status || DEFAULT_TEXT },
-      // { 'Incumbent': data?.status || DEFAULT_TEXT },
-      // { 'Tour of Duty': data?.status || DEFAULT_TEXT },
-      // { 'Assignee': data?.status || DEFAULT_TEXT },
-      // { 'Post Differential | Danger Pay': data?.status || DEFAULT_TEXT },
-    ],
+    bodySecondary: PP_INTEGRATION_FLAG ?
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:44]: 🙌

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 7]
```diff
@@ -63,14 +95,14 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
...
-    // re-reading from "pos" when we open Edit Form back up
-    // clear will need to set states back to the pull
-    // from "pos" once we've determined the ref data structure
-    const originalText = data?.positionDetails ? data.positionDetails : 'No description';
-    setTextArea(originalText);
-    setEditMode(false);
+    setStatus({});
+    setExclude(true);
+    setSelectedCycles([]);
+    setTextArea(data?.positionDetails || 'No description.');
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:101]: 🙌

File: src/Components/Assignments/NotificationCard/Tabs/Assignments.jsx (hunk 5 above)
Comment (elizabeth-jimenez) [reply src/Components/Assignments/NotificationCard/Tabs/Assignments.jsx:41]: rip

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 8]
```diff
@@ -45,6 +61,22 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
...
+  const pickyProps = {
+    numberDisplayed: 2,
+    multiple: true,
+    includeFilter: true,
+    dropdownHeight: 200,
+    includeSelectAll: true,
+    renderList: renderSelectionList,
+    className: 'width-280',
+  };
+  const [status, setStatus] = useState({});
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:73]: doesn't matter for UI, but should be  `useState('')`

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 9]
```diff
@@ -63,14 +95,14 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
...
 
   const onCancelForm = () => {
-    // this is likely not going to be needed, as we should be
-    // re-reading from "pos" when we open Edit Form back up
-    // clear will need to set states back to the pull
-    // from "pos" once we've determined the ref data structure
-    const originalText = data?.positionDetails ? data.positionDetails : 'No description';
-    setTextArea(originalText);
-    setEditMode(false);
+    setStatus({});
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:98]: same as above `('')`

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 10]
```diff
@@ -83,6 +115,47 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
     ],
     inputBody: (
       <div className="position-form">
+        { PP_INTEGRATION_FLAG ?
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:118]: ternary works, but we could also do `PP_INTEGRATION_FLAG &&`

File: public/config/config.json [hunk 11]
```diff
@@ -31,6 +31,7 @@
     "post_handshake": false,
     "projected_vacancy": true,
     "publishable_positions": true,
+    "publishable_positions_integration": true,
```
Comment (elizabeth-jimenez) [inline public/config/config.json:34]: flags cleanly removing Edit functionality 🙌

File: src/Components/PublishablePositionCard/PublishablePositionCard.jsx [hunk 12]
```diff
@@ -135,16 +304,25 @@ const PublishablePositionCard = ({ data, onEditModeSearch, onSubmit, disableEdit
           sections={sections}
           form={form}
         />,
+      }, {
+        text: PP_INTEGRATION_FLAG ? 'Position Classification' : '',
+        value: PP_INTEGRATION_FLAG ? 'CLASSIFICATION' : '',
+        content: PP_INTEGRATION_FLAG ? classificationTable() : null,
+        disabled: editMode,
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositionCard/PublishablePositionCard.jsx:311]: I think we may be able to do this here

```
PP_INTEGRATION_FLAG ? {
        text: 'Position Classification',
        value: 'CLASSIFICATION',
        content: classificationTable(),
        disabled: editMode,
      } : {}
```

File: src/Components/PublishablePositions/PublishablePositions.jsx [hunk 13]
```diff
@@ -34,34 +37,38 @@ const PublishablePositions = ({ viewType }) => {
 
   const [tempsearchPosNum, tempsetSearchPosNum] = useState(userSelections?.searchPosNum || '');
   const [searchPosNum, setSearchPosNum] = useState(userSelections?.searchPosNum || '');
-  const [selectedStatuses, setSelectedStatuses] = useState(userSelections?.selectedStatus || []);
+  const [selectedStatuses, setSelectedStatuses] = useState(userSelections?.selectedStatuses || []);
   const [selectedBureaus, setSelectedBureaus] = useState(userSelections?.selectedBureaus || []);
   const [selectedOrgs, setSelectedOrgs] = useState(userSelections?.selectedOrgs || []);
-  const [selectedGrades, setSelectedGrades] = useState(userSelections?.selectedGrade || []);
+  const [selectedGrades, setSelectedGrades] = useState(userSelections?.selectedGrades || []);
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositions/PublishablePositions.jsx:43]: 💅

File: src/Components/PublishablePositions/PublishablePositions.jsx [hunk 14]
```diff
@@ -34,34 +37,38 @@ const PublishablePositions = ({ viewType }) => {
...
   const skillOptions = uniqBy(sortBy(skills, [(f) => f.description]), 'code');
   const orgOptions = uniqBy(sortBy(orgs, [(f) => f.description]), 'code');
+  const cycleOptions = uniqBy(sortBy(cycles, [(f) => f.code]), 'code');
   const gradeOptions = uniqBy(grades, 'code');
 
   const getQuery = () => ({
     posNum: searchPosNum,
     statuses: selectedStatuses.map(f => (f?.code)),
-    bureaus: selectedBureaus.map(f => (f?.description)),
+    bureaus: selectedBureaus.map(f => (f?.code)),
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositions/PublishablePositions.jsx:67]: intentionally `description`

**Comment by M00staff** [comment]
should be GTG

File: src/Components/PublishablePositions/PublishablePositions.jsx (hunk 13 above)
Comment (elizabeth-jimenez) [reply src/Components/PublishablePositions/PublishablePositions.jsx:43]: nice fixes - this is why it was behaving a little strange!

File: src/Components/PublishablePositions/PublishablePositions.jsx [hunk 15]
```diff
@@ -129,7 +139,9 @@ const PublishablePositions = ({ viewType }) => {
     setSelectedOrgs([]);
     setSelectedGrades([]);
     setSelectedSkills([]);
+    setSelectedBidCycles([]);
     setClearFilters(false);
+    dispatch(savePublishablePositionsSelections({}));
```
Comment (elizabeth-jimenez) [inline src/Components/PublishablePositions/PublishablePositions.jsx:144]: Nice catch!

**Pull request merged**